glib/src/subclass/signal.rs

```diff
@@ -68,9 +69,117 @@ impl fmt::Debug for SignalInvocationHint {
     }
 }
 
+/// In-depth information of a specific signal
+#[repr(transparent)]
+pub struct SignalQuery(gobject_ffi::GSignalQuery);
+
+impl SignalQuery {
+    /// The name of the signal.
+    pub fn signal_name(&self) -> GString {
+        unsafe {
+            let name: GString = from_glib_none(self.0.signal_name);
+            name
+        }
+    }
+
+    /// The ID of the signal.
+    pub fn signal_id(&self) -> SignalId {
+        unsafe { SignalId::new(NonZeroU32::new_unchecked(self.0.signal_id)) }
+    }
+
+    /// The interface/instance this signal can be emitted for.
+    pub fn type_(&self) -> Type {
+        unsafe { from_glib(self.0.itype) }
+    }
+
+    /// The signal flags.
+    pub fn flags(&self) -> SignalFlags {
+        unsafe { from_glib(self.0.signal_flags) }
+    }
+
+    /// The return type for the user callback.
+    pub fn return_type(&self) -> Type {
+        unsafe { from_glib(self.0.return_type) }
+    }
+
+    /// The number of parameters the user callback takes.
+    pub fn n_params(&self) -> u32 {
+        self.0.n_params
+    }
+
+    /// The parameters for the user callback.
+    pub fn param_types(&self) -> Vec<Type> {
```

Comment:
The SignalBuilder code uses arg_types & ret_types, I wonder if we should use the same names here? :)

Reply:
Yes let's make it consistent. Unfortunately it's also not consistent in the C API though.

I would go with param/parameter instead of arg/argument, apart from that select something you like :)

Reply:
Might also make sense to use a `SmallVec<[Type; 16]>` or something like that here. We depend on smallvec anyway and this will basically never be more than 10 or so :)